### PR TITLE
feat: define remote table engine trait

### PR DESCRIPTION
**Pull request by Rachelint**
# Which issue does this PR close?

Closes #
Part of #492 

# Rationale for this change
 See #492.
We need to read/write the remote table engine.

# What changes are included in this PR?
+ define the remote table engine trait

# Are there any user-facing changes?
None.

# How does this change test
Not need now.

### CODE REVIEW

File: remote_engine/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+[package.version]
+workspace = true
+
+[package.authors]
+workspace = true
+
+[package.edition]
+workspace = true
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (ShiKaiWi) [inline remote_engine/Cargo.toml:12]: Maybe we can remove this.

File: remote_engine/src/mock_impl.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+//! Mock impl for remote table engine
+
+use async_trait::async_trait;
+pub struct MockImpl;
+
+#[async_trait]
+impl RemoteEngine for MockImpl {
```
Comment (ShiKaiWi) [inline remote_engine/src/mock_impl.rs:9]: Maybe this trait should be defined in the `table_engine` crate?

File: remote_engine/src/model.rs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+//! Model for remote table engine
+
+use table_engine::table::{ReadRequest as TableReadRequest, WriteRequest as TableWriteRequest};
+
+#[allow(dead_code)]
+pub struct FullTableName {
```
Comment (ShiKaiWi) [inline remote_engine/src/model.rs:8]: `TableIdentifier` seems a better name.

File: remote_engine/src/mock_impl.rs (hunk 2 above)
Comment (Rachelint) [reply remote_engine/src/mock_impl.rs:9]: Really a good idea.

**Review by ShiKaiWi — APPROVED** [review]
LGTM

**Pull request merged**